### PR TITLE
[GEOS-11541] STAC search endpoint sortby query not working with POST

### PR DESCRIPTION
**Pull request by turingtestfail**
[GEOS-11541](https://osgeo-org.atlassian.net/browse/GEOS-11541) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/v1/stac/STACService.java [hunk 1]
```diff
@@ -525,6 +537,46 @@ public SearchResponse searchPost(@RequestBody STACSearchQuery sq) throws Excepti
 
         return response;
     }
+    /**
+     * Based on STAC API Extension for Sort
+     * https://github.com/stac-api-extensions/sort?tab=readme-ov-file#http-post-json-entity *
+     */
+    private SortBy[] stacSortByToGTSortby(SortBy[] sortBy) {
```
Comment (aaime) [inline src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/v1/stac/STACService.java:544]: I'd rather have the custom parsing in ``STACSearchQuery#setSortBy(JsonNode)``

Comment (turingtestfail) [reply src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/v1/stac/STACService.java]: Done

File: src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/v1/stac/STACSearchQuery.java [hunk 2]
```diff
@@ -46,4 +58,37 @@ public Integer getStartIndexAsInt() {
...
+    public void setSortBy(JsonNode node) {
+        // Based on STAC API Extension for Sort
+        // https://github.com/stac-api-extensions/sort?tab=readme-ov-file#http-post-json-entity
+        if (node instanceof ArrayNode) {
+            ArrayNode arrayNode = (ArrayNode) node;
+            SortBy[] sortBIES = new SortBy[arrayNode.size()];
+            for (int i = 0; i < arrayNode.size(); i++) {
+                if (arrayNode.get(i).isObject()
+                        && arrayNode.get(i).has(STAC_SORTBY_FIELD)
+                        && arrayNode.get(i).has(STAC_SORTBY_DIRECTION)) {
```
Comment (aaime) [inline src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/v1/stac/STACSearchQuery.java:72]: I find having a mandatory direction odd, and yet the schema in the spec has it as mandatory, indeed.
[I've asked for a clarification](https://github.com/stac-api-extensions/sort/issues/10), in the meantime, this is correct.

**Review by aaime — APPROVED** [review]

**Comment by jodygarnett** [comment]
main frozen please do not merge stuff see developer list.

**Pull request merged**